chord_metadata_service/phenopackets/api_views.py

```diff
@@ -170,6 +173,46 @@ class BiosampleViewSet(ExtendedPhenopacketsModelViewSet):
     queryset = m.Biosample.objects.all().prefetch_related(*BIOSAMPLE_PREFETCH).order_by("id")
 
 
+class BiosampleBatchViewSet(ExtendedPhenopacketsModelViewSet):
+    """
+    get:
+    Return a list of all existing biosamples
+
+    post:
+    Filter biosamples by a list of ids
+    """
+    serializer_class = s.BiosampleSerializer
+    filter_backends = [DjangoFilterBackend]
+    filterset_class = f.BiosampleFilter
+    pagination_class = BatchResultsSetPagination
+    renderer_classes = (*api_settings.DEFAULT_RENDERER_CLASSES, FHIRRenderer,
+                        PhenopacketsRenderer, BiosamplesCSVRenderer, ARGORenderer,
+                        IndividualBentoSearchRenderer)
+    content_negotiation_class = FormatInPostContentNegotiation
+
+    def _get_filtered_queryset(self, ids_list=None):
+        queryset = m.Biosample.objects.all() \
+            .prefetch_related(*BIOSAMPLE_PREFETCH) \
+            .select_related(*BIOSAMPLE_SELECT_REL) \
+            .order_by("id")
+
+        if ids_list:
+            queryset = queryset.filter(id__in=ids_list)
```

Comment:
I think we should apply this filter before `prefetch_related` and `select_related` to limit db operations when ids are passed.

Reply:
Absolutely, you're correct. Thank you for pointing this out!